src/python/gem5/utils/multisim/multisim.py

```diff
@@ -200,17 +206,53 @@ def run(module_path: Path, processes: Optional[int] = None) -> None:
         "(after determining number of jobs)."
     )
 
-    # Setup the multiprocessing pool. If the number of processes is not
-    # specified (i.e. `None`) the default is the number or available threads.
-    from ..multiprocessing.context import gem5Context
-
-    pool = gem5Context().Pool(processes=max_num_processes, maxtasksperchild=1)
-
-    # Use the starmap function to create N child processes each with same
-    # module path (the config script specifying all simulations using MultiSim)
-    # but a different ID. The ID is used to select the correct simulator to
-    # run.
-    pool.starmap(_run, zip([module_path for _ in range(len(ids))], tuple(ids)))
+    active_processes = []
+    remaining_ids = list(ids).copy()
+    process_lock = Lock()
+    from gem5.utils.multiprocessing import Process
+
+    def handle_exit(signum, frame):
+        """Signal handler to clean up processes on termination."""
+        import sys
+
+        inform("Cleaning up processes")
+        with process_lock:
+            for process in active_processes:
+                if process.is_alive():
+                    inform(f"Terminating process {process.name}")
+                    process.terminate()
+        sys.exit(0)
+
+    # Register signal handler
+    signal.signal(signal.SIGINT, handle_exit)
+    signal.signal(signal.SIGTERM, handle_exit)
+
+    try:
+        while remaining_ids or active_processes:
+            # Start new processes if available
+            while remaining_ids and len(active_processes) < max_num_processes:
+                id_to_run = remaining_ids.pop()
+                try:
+                    process = Process(
+                        target=_run,
+                        args=(module_path, id_to_run),
+                        name=id_to_run,
+                    )
+                    process.start()
+                    with process_lock:
+                        active_processes.append(process)
+                except Exception as e:
+                    inform(f"Error starting process for {id_to_run}: {e}")
+            # Wait for active processes to finish
+            time.sleep(1)  # Avoid busy-waiting
+            with process_lock:
+                active_processes = [
```

Comment:
Why the "copy" here? I have a gut feeling why you chose to do this, but it would be good to document in a comment why you made that decision.

Reply:
Just to clarify, you mean why I used `.copy` to initialize the variable?

If yes that was so that the original list is not modified.

Reply:
No, I mean that you use a list comprehension instead of directly modifying `active_processes`